src/widget.cpp

```diff
@@ -2992,6 +3000,12 @@ void NWidgetLeaf::Draw(const Window *w)
 			DrawFrameRect(r.left, r.top, r.right, r.bottom, this->colour, (clicked) ? FrameFlag::Lowered : FrameFlags{});
 			break;
 
+		case WWT_BOOLBTN: {
+			Point pt = GetAlignedPosition(r, Dimension(SETTING_BUTTON_WIDTH, SETTING_BUTTON_HEIGHT), this->align);
+			DrawBoolButton(pt.x, pt.y, COLOUR_YELLOW, this->colour, clicked, !this->IsDisabled());
```

Comment:
Should not it be
```suggestion
			DrawBoolButton(pt.x, pt.y, this->colour, this->colour, clicked, !this->IsDisabled());
```
?

Yellow slider when all other buttons are grey in the window looks wrong to me.

Reply:
Basically this is a precursor to #13242 where the other buttons will be yellow. But, that assumes that'll be merged.

Using `this->colour` is wrong here for all the other cases where the button should be yellow. A widget can't currently have two colours, so to allow this to be properly flexible would require adding a Colours to WidgetData and using that. And I didn't do this as I hope it's not going to be needed very long.

But on the other hand, we might want to use this elsewhere with non-yellow buttons, so maybe I should do this anyway.